homeassistant/components/tolosauna/__init__.py

```diff
@@ -0,0 +1,125 @@
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DEFAULT_RETRY_COUNT, DEFAULT_RETRY_TIMEOUT, DOMAIN
+
+PLATFORMS = ["climate"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up tolosauna from a config entry."""
+    client = ToloClient(entry.data[CONF_HOST])
+    coordinator = ToloSaunaUpdateCoordinator(hass, entry)
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {"client": client, "coordinator": coordinator}
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class ToloSaunaData(NamedTuple):
+    """Compound class for reflecting full state (status and info) of a TOLO Sauna."""
+
+    status: StatusInfo
+    settings: SettingsInfo
+
+
+class ToloSaunaUpdateCoordinator(DataUpdateCoordinator[ToloSaunaData]):
+    """DataUpdateCoordinator for TOLO Sauna."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize ToloSaunaUpdateCoordinator."""
+        self._config_entry = entry
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=f"{entry.title} ({entry.data[CONF_HOST]}) Data Update Coordinator",
+            update_interval=timedelta(seconds=3),
+        )
+
+    async def _async_update_data(self) -> ToloSaunaData:
+        client = ToloClient(self._config_entry.data[CONF_HOST])
+        try:
+            status = client.get_status_info(
+                resend_timeout=DEFAULT_RETRY_TIMEOUT, retries=DEFAULT_RETRY_COUNT
+            )
+            settings = client.get_settings_info(
+                resend_timeout=DEFAULT_RETRY_TIMEOUT, retries=DEFAULT_RETRY_COUNT
+            )
+            return ToloSaunaData(status, settings)
+        except ResponseTimedOutError as error:
+            raise UpdateFailed("communication timeout") from error
+
+
+class ToloSaunaCoordinatorEntity(CoordinatorEntity):
+    """CoordinatorEntity for TOLO Sauna."""
+
+    coordinator: ToloSaunaUpdateCoordinator
+
+    def __init__(
+        self, coordinator: ToloSaunaUpdateCoordinator, entry: ConfigEntry
+    ) -> None:
+        """Initialize ToloSaunaCoordinatorEntity."""
+        self._config_entry = entry
+        super().__init__(coordinator)
+
+    @property
+    def status(self) -> StatusInfo:
+        """Return TOLO Sauna status info."""
+        return self.coordinator.data.status
+
+    @property
+    def settings(self) -> SettingsInfo:
+        """Return TOLO Sauna settings info."""
+        return self.coordinator.data.settings
```

Comment:
This is all is already typed and such, any reason to add these proxy methods?

Reply:
Just for convenience reasons, I will remove them.